3-extra/card-validation.js

```diff
@@ -0,0 +1,10 @@
+function onlyNumbers(string) {
+  return /^[0-9]*$/.test(string);
+}
+// check if numbers only / sum > 16 / check if all numbers are same / check if last digit is even
+const ccValidator = (cc) => {
+  const sum = cc.split("").reduce((a, i) => a + parseInt(i, 10), 0);
+  return onlyNumbers(cc) && sum >= 16 && sum / 16 !== parseInt(cc[0], 10) && cc[15] % 2 === 0;
+}
+
+console.log(ccValidator("6666666666666661"));
```

Comment:
Did you find this exercise hard? Just looking through your function, it's pretty hard to tell at a glance what rules are being checked on line 7. Here's mine:
```javascript
function validateCreditCardNumber(cardNumber) {
  // - Number must be 16 digits, all of them must be numbers.
  if (!cardNumber.match(/^\d{16}$/)) return false;
  // - You must have at least two different digits represented (all of the digits cannot be the same).
  if (cardNumber === "".padStart(16, cardNumber[0])) return false;
  // - The final digit must be even.
  if (!cardNumber.match(/[02468]$/)) return false;
  // - The sum of all the digits must be greater than 16.
  if (
    cardNumber.split("").reduce((total, char) => (total += Number(char)), 0) <=
    16
  )
    return false;
  // Must be OK
  return true;
}
```
I left the comments in for the validation rules and used some regular expressions to simplify things.